event-app-ui/src/components/Popup/Popup.jsx

```diff
@@ -15,21 +17,105 @@ export default function Popup(props) {
   const [location, setLocation] = useState("");
   const [startTime, setStartTime] = useState("");
   const [endTime, setEndTime] = useState("");
+  const [users, setUsers] = useState([]);
+  const [selectedUsers, setSelectedUsers] = useState([]);
+  const [gaps, setGaps] = useState([]);
+  const [error, setError] = useState();
+  const [isChecked, setIsChecked] = useState(false);
 
-  const { setIsOpen } = props;
-  const { username } = props;
+  const dateConversionConfig = {
+    year: "2-digit",
+    month: "2-digit",
+    day: "2-digit",
+    hour: "numeric",
+    minute: "numeric",
+    hour12: true,
+  };
+
+  const { username, setIsOpen } = props;
 
-  const eventInput = () => {
-    axios.post(`http://localhost:3000/event_popup`, {
+  const eventInput = async (e) => {
+    const response = await axios.post(`http://localhost:3000/event_popup`, {
       eventName: eventName,
       description: description,
       location: location,
       startTime: startTime,
       endTime: endTime,
       username: username,
+      selectedUsers: selectedUsers,
     });
+    //sends an
+    if (response.data.errorMessage) {
+      e.preventDefault();
+      setError(response.data.errorMessage);
+    } else {
+      setIsOpen(false);
+      setError("");
```

Comment:
maybe you can set detault error to empty string or null so that you dont need to set explicitly incase of no error